File: api/wmb_api/app/models/user_models.py
from sqlalchemy import Column, String, DateTime, ForeignKey, BigInteger
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from ..config.database import Base
from sqlalchemy.dialects.mysql import BIGINT


class User(Base):
    __tablename__ = "users"
    
    id = Column(BIGINT(unsigned=True), primary_key=True, index=True, autoincrement=True)
    name = Column(String(255), nullable=False)
    email = Column(String(255), unique=True, nullable=False, index=True)
    id_google = Column(String(255), nullable=True)
    email_verified_at = Column(DateTime, nullable=True)
    password = Column(String(255), nullable=False)
    remember_token = Column(String(255), nullable=True)
    created_at = Column(DateTime, default=func.now())
    updated_at = Column(DateTime, default=func.now(), onupdate=func.now())
    otp_code = Column(String(6), nullable=True)
    otp_expires_at = Column(DateTime, nullable=True)
    last_login = Column(DateTime, nullable=True)
    
    user_images = relationship("UserImage", back_populates="user", cascade="all, delete-orphan")

class UserImage(Base):
    __tablename__ = "user_images"
    
    id = Column(BIGINT(unsigned=True), primary_key=True, index=True, autoincrement=True)
    users_id = Column(BIGINT(unsigned=True), ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
    gambar = Column(String(255), nullable=True)
    created_at = Column(DateTime, default=func.now())
    updated_at = Column(DateTime, default=func.now(), onupdate=func.now())
    
    user = relationship("User", back_populates="user_images")

File: express_api/src/models/userModels.js
// models/userModels.js

import { DataTypes } from 'sequelize'; 
import sequelize from '../config/db.js'; 
import { Op } from 'sequelize';

// relasi import
import {AlamatUser} from './alamatUserModels.js';

// Model User
const User = sequelize.define(
  'User',
  {
    id: {
      type: DataTypes.BIGINT,
      autoIncrement: true,
      primaryKey: true,
    },
    name: {
      type: DataTypes.STRING(255),
      allowNull: false,
    },
    email: {
      type: DataTypes.STRING(255),
      allowNull: false,
      unique: true,
    },
    id_google: {
      type: DataTypes.STRING(255),
      allowNull: true,
    },
    email_verified_at: {
      type: DataTypes.DATE,
      allowNull: true,
    },
    password: {
      type: DataTypes.STRING(255),
      allowNull: false,
    },
    remember_token: {
      type: DataTypes.STRING(255),
      allowNull: true,
    },
    created_at: {
      type: DataTypes.DATE,
      allowNull: true,
      defaultValue: DataTypes.NOW,
    },
    updated_at: {
      type: DataTypes.DATE,
      allowNull: true,
      defaultValue: DataTypes.NOW,
    },
    otp_code: {
      type: DataTypes.STRING(6),
      allowNull: true,
    },
    otp_expires_at: {
      type: DataTypes.DATE,
      allowNull: true,
    },
    last_login: {
      type: DataTypes.DATE,
      allowNull: true,
    },
  },
  {
    tableName: 'users',
    timestamps: true,
    underscored: true,
  }
);

// Model GambarUser
const GambarUser = sequelize.define(
  'GambarUser',
  {
    id: {
      type: DataTypes.BIGINT,
      autoIncrement: true,
      primaryKey: true,
    },
    users_id: {
      type: DataTypes.BIGINT,
      allowNull: false,
    },
    gambar: {
      type: DataTypes.STRING(255),
      allowNull: true,
    },
    created_at: {
      type: DataTypes.DATE,
      allowNull: true,
      defaultValue: DataTypes.NOW,
    },
    updated_at: {
      type: DataTypes.DATE,
      allowNull: true,
      defaultValue: DataTypes.NOW,
    },
  },
  {
    tableName: 'gambar_user',
    timestamps: true,
    underscored: true,
  }
);

// Relasi
User.hasMany(GambarUser, { foreignKey: 'users_id', as: 'gambar' });
GambarUser.belongsTo(User, { foreignKey: 'users_id', as: 'user' });
User.hasMany(AlamatUser, { foreignKey: 'user_id', as: 'alamat' });

// Ekspor model
export { User, GambarUser };